runtime/src/bank.rs

```diff
@@ -9301,28 +9257,26 @@ pub(crate) mod tests {
         // verify that there's inflation
         assert_ne!(bank1.capitalization(), bank0.capitalization());
 
-        // verify the inflation is represented in validator_points *
-        let paid_rewards = bank1.capitalization()
-            - bank0.capitalization()
-            - bank1_sysvar_delta()
-            - new_epoch_sysvar_delta();
+        // verify the inflation is represented in validator_points
+        let paid_rewards = bank1.capitalization() - bank0.capitalization() - bank1_sysvar_delta();
 
-        let rewards = bank1
-            .get_account(&sysvar::rewards::id())
-            .map(|account| from_account::<Rewards, _>(&account).unwrap())
-            .unwrap();
+        let slot_in_year = bank1.slot_in_year_for_inflation();
+        let epoch_duration_in_years = bank1.epoch_duration_in_years(bank0.epoch());
+        let validator_rate = bank1.inflation.read().unwrap().validator(slot_in_year);
+
+        let allocated_rewards =
+            (validator_rate * bank0.capitalization() as f64 * epoch_duration_in_years) as f64;
```

Comment:
This looks good but do you mind adding a helper function to the bank which calculates the allocated rewards and is called by `update_rewards_with_thread_pool` and this test?

Reply:
Added `inflation_allocated_during_epoch ` to bundle this logic together. One bummer is that I couldn't remove much code in `update_rewards_with_thread_pool ` because it is logging a bunch of the intermediate variables